### PR TITLE
[prometheus-kafka-exporter] Add support for securityContext

### PR DESCRIPTION
**Pull request by zeritti**
Signed-off-by: zeritti <47476160+zeritti@users.noreply.github.com>

#### What this PR does / why we need it
This PR adds support for `securityContext` in the kafka-exporter's container enabling thus transition from PSP to Pod Security Standards. 

Since value `securityContext` is already used for pod's security context, `containerSecurityContext` has been chosen for that of the container.

#### Which issue this PR fixes
None

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-kafka-exporter/values.yaml [hunk 1]
```diff
@@ -143,8 +143,19 @@ server:
...
 
-# Set security context for the kafka-exporter container. Useful when you need to adapt to an existing PSP
+# Set security context for the kafka-exporter pod. Useful when you need to adapt to an existing PSP
 securityContext: {}
   # runAsUser:
   # runAsGroup:
   # fsGroup:
+
+# Set securityContext for the kafka-exporter container
+containerSecurityContext: {}
```
Comment (monotek) [inline charts/prometheus-kafka-exporter/values.yaml:153]: please sort examples alphabetically.

File: charts/prometheus-kafka-exporter/values.yaml [hunk 2]
```diff
@@ -143,8 +143,19 @@ server:
 kafka:
   enabled: false
 
-# Set security context for the kafka-exporter container. Useful when you need to adapt to an existing PSP
+# Set security context for the kafka-exporter pod. Useful when you need to adapt to an existing PSP
 securityContext: {}
   # runAsUser:
```
Comment (monotek) [inline charts/prometheus-kafka-exporter/values.yaml:148]: these examples hsould have some example values too.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-kafka-exporter/values.yaml (hunk 2 above)
Comment (zeritti) [reply charts/prometheus-kafka-exporter/values.yaml]: Example values added.

File: charts/prometheus-kafka-exporter/values.yaml (hunk 1 above)
Comment (zeritti) [reply charts/prometheus-kafka-exporter/values.yaml:155]: Examples sorted.

File: charts/prometheus-kafka-exporter/values.yaml [hunk 3]
```diff
@@ -145,17 +145,20 @@ kafka:
...
-  # runAsUser: 10001
-  # runAsNonRoot: true
   # allowPrivilegeEscalation: false
   # capabilities:
   #   drop: ["all"]
   # readOnlyRootFilesystem: true
+  # runAsGroup: 10002
+  # runAsNonRoot: true
+  # runAsUser: 10001
   # seccompProfile:
```
Comment (monotek) [inline charts/prometheus-kafka-exporter/values.yaml:163]: "seccompProfile" should not work in containercontext

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-kafka-exporter/values.yaml (hunk 3 above)
Comment (zeritti) [reply charts/prometheus-kafka-exporter/values.yaml:163]: It works in both contexts as per API spec [securitycontext](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#securitycontext-v1-core).

**Review by monotek — APPROVED** [review]

**Pull request merged**